### PR TITLE
Persist current Ethereum network

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/17641

- Added preference to keep current chain id

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
- Go to wallet and change the active network from mainnet to any other
- Restart wallet and check if selected network is still active

### CODE REVIEW

**Comment by darkdh** [comment]
Just a reminder, do not merge any PRs including this one into https://github.com/brave/brave-core/pull/9798 again.

File: components/brave_wallet/browser/eth_json_rpc_controller.cc [hunk 1]
```diff
@@ -244,6 +258,7 @@ void EthJsonRpcController::OnGetTransactionReceipt(
 // static
 void EthJsonRpcController::RegisterProfilePrefs(PrefRegistrySimple* registry) {
   registry->RegisterListPref(kBraveWalletCustomNetworks);
+  registry->RegisterStringPref(kBraveWalletCurrentChainId, std::string());
```
Comment (darkdh) [inline components/brave_wallet/browser/eth_json_rpc_controller.cc:261]: just register the default value here instead of using fallback function

Comment (spylogsster) [reply components/brave_wallet/browser/eth_json_rpc_controller.cc:261]: the fallback is used not only to initialize pref, but also to provide fallback if user adds some broken network

Comment (darkdh) [reply components/brave_wallet/browser/eth_json_rpc_controller.cc:261]: so why didn't we prevents users from adding broken networks?

Comment (spylogsster) [reply components/brave_wallet/browser/eth_json_rpc_controller.cc:261]: we have issue about it https://github.com/brave/brave-browser/issues/17640, hope we can add this soon

Comment (darkdh) [reply components/brave_wallet/browser/eth_json_rpc_controller.cc]: what I'm trying to say is, you didn't check broken networks in this PR so basically this fallback only serves as getting default value

File: components/brave_wallet/browser/eth_json_rpc_controller.cc [hunk 2]
```diff
@@ -306,6 +307,8 @@ void EthJsonRpcController::OnGetTransactionReceipt(
 // static
 void EthJsonRpcController::RegisterProfilePrefs(PrefRegistrySimple* registry) {
   registry->RegisterListPref(kBraveWalletCustomNetworks);
+  registry->RegisterStringPref(kBraveWalletCurrentChainId,
```
Comment (darkdh) [inline components/brave_wallet/browser/eth_json_rpc_controller.cc:310]: we also need a static function to clear the prefs when `KeyringController::Reset()`

Comment (spylogsster) [reply components/brave_wallet/browser/eth_json_rpc_controller.cc:310]: done

**Review by darkdh — APPROVED** [review]

**Pull request merged**